packages/create-amplify/src/package_manager.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  NpmProjectInitializer,
+  PnpmProjectInitializer,
+  YarnClassicProjectInitializer,
+  YarnModernProjectInitializer,
+} from './project-initializer/index.js';
+import {
+  NpmPackageManagerController,
+  PnpmPackageManagerController,
+  YarnClassicPackageManagerController,
+  YarnModernPackageManagerController,
+} from './package-manager-controller/index.js';
+
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = Readonly<string[]>;
+export type PackageManager = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
```

Comment:
I think `ProjectInitializer` and `PackageManagerController` could become abstract classes and we could have:

```
type PackageManagerProps: {
  name: string, executable: string, binaryRunner: string, lockFileName:string,
}

abstract class PackageManagerController {
 constructor(props: PackageManagerProps)

  //rest of interface
}

abstract class ProjectInitializer {
 constructor(props: PackageManagerProps)

  //rest of interface
}
```

We should also remove types above. These values matter only for subtypes and should become implementation details.

Reply:
Done [7ec208b](https://github.com/aws-amplify/amplify-backend/pull/793/commits/7ec208b395ec0cc6151c5e7031510b6c816fdd62)